docs/API.md

```diff
@@ -812,6 +813,57 @@ This will print something like this:
 
 When an assertion fails, the error message should give as much signal as necessary to the user so they can resolve their issue quickly. It's usually recommended to spend a lot of time crafting a great failure message to make sure users of your custom assertions have a good developer experience.
 
+### Asymmetric Jest Matchers
+
+Sometimes you don't want to check equality of entire object. You just need to assert that value is not empty or has some expected type. For example, we want to check the shape of some message entity:
+
+```
+  expect({
+    timestamp: 1480807810388,
+    text: 'Some text content, but we care only about *this part*'
+  }).toEqual({
+    timestamp: expect.any(Number),
+    text: expect.strMatch('*this part*')
+  });
+```
+
+There some special values with specific comparing behavior that you can use as a part of expectation. They are useful for asserting some types of data, like timestamps, or long text resources, where only part of it is important for testing. Currently, Jest has the following asymmetric matches:
+  
+  * `expect.anything()` - matches everything, except `null` and `undefined`
+  * `expect.any(<constructor>)` - checks, that actual value is instance of provided `<constructor>`.
+  * `expect.objectContaining(<object>)` - compares only keys, that exist in provided object. All other keys of `actual` value will be ignored.
+  * `expect.objPart(<object>)` - short alias for `expect.objectContaining`
```

Comment:
As commented below, I'm not a fan of abbreviations in API design and it diverges from the Jasmine API. I feel pretty strongly about this.

I'm happy to either:
* Accept the jasmine API names as the only way forward but not introduce additional aliases.
* Find good aliases for the Jasmine API that are as clear as the current names, without abbreviating "object" to "obj" etc.